src/de/mod.rs

```diff
@@ -19,6 +19,7 @@ pub type Result<T> = core::result::Result<T, Error>;
 /// This type represents all possible errors that can occur when deserializing JSON data
 #[derive(Debug, PartialEq, Eq, Clone)]
 #[cfg_attr(not(feature = "custom-error-messages"), derive(Copy))]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
```

Comment:
I think you need to add:
```rs
        #[defmt(Debug2Format)]
```

above the `heapless::String<64>` error type.

Reply:
or adding "defmt-impl" into the heapless dependency